### PR TITLE
Add support for local variables

### PR DESCRIPTION
**Pull request by JanJecmen**

### CODE REVIEW

File: rir/src/interpreter/interp_context.h [hunk 1]
```diff
@@ -158,6 +158,48 @@ INLINE void ostack_ensureSize(Context* c, unsigned minFree) {
...
+        assert(offset < localsCount && "Attempt to store invalid local variable.");
+#ifdef TYPED_STACK
+        (base + offset)->u.sxpval = val;
+        (base + offset)->tag = 0;
+#else
+        base[offset] = val;
+#endif
+    }
+
+    // maybe this is an overkill?
```
Comment (o-) [inline rir/src/interpreter/interp_context.h]: nono, I agree :)

File: rir/src/runtime/Function.h [hunk 2]
```diff
@@ -47,21 +47,22 @@ typedef SEXP FunctionSEXP;
 #pragma pack(push)
 #pragma pack(1)
 struct Function {
+public:
```
Comment (o-) [inline rir/src/runtime/Function.h]: indentation seems off, maybe your clang format somehow failed to run?

**Review by o- — APPROVED** [review]
looks good to me

File: rir/src/runtime/Function.h (hunk 2 above)
Comment (JanJecmen) [reply rir/src/runtime/Function.h]: yes, it failed :) `<q><n><enter>`
sorry, will start to use it

**Comment by JanJecmen** [comment]
As my atonement, I have run clang-format by hand for the files in this PR. Should look better now.

**Pull request merged**